v2.0/configure-replication-zones.md

```diff
@@ -73,8 +73,15 @@ The node-level and store-level descriptive attributes mentioned above can be use
 
 Constraint Type | Description | Syntax
 ----------------|-------------|-------
-**Required** | When placing replicas, the cluster will consider only nodes/stores with matching attributes. When there are no matching nodes/stores with capacity, new replicas will not be added. | `[+ssd]`
-**Prohibited** | When placing replicas, the cluster will ignore nodes/stores with matching attributes. When there are no alternate nodes/stores with capacity, new replicas will not be added. | `[-ssd]`
+**Required** | When placing replicas, the cluster will consider only nodes/stores with matching attributes or localities. When there are no matching nodes/stores, new replicas will not be added. | `+ssd`
+**Prohibited** | When placing replicas, the cluster will ignore nodes/stores with matching attributes or localities. When there are no alternate nodes/stores, new replicas will not be added. | `-ssd`
+
+Constraints can be specified such that they apply to all replicas in a zone or such that different constraints apply to different replicas, meaning you can effectively pick the exact location of each replica.
+
+Constraint Scope | Description | Syntax
+----------------|-------------|-------
+**All Replicas** | Constraints specified using list syntax apply to all replicas in every range that's part of the replication zone. | `constraints: [+ssd, -region=west]`
+**Per-Replica** | Multiple lists of constraints can be provided in a map from the list of constraints to an integer number of replicas in each range that the constraints should apply to. The total number of replicas constrained cannot be greater than the configured number of replicas for the zone. | `constraints: {"+ssd,-region=west": 2, "+region=east": 1}`
```

Comment:
Similar to the above, perhaps we should change `provided in a map from the list of constraints to...` to `provided in a JSON object mapping the list of constraints to...`

Reply:
Agreed.

Reply:
Done.